### PR TITLE
adding support for multifile read_only, refactor in read-only styles and fixing glitch in multifile editor

### LINKED ISSUE #1146: Sometimes the code editor is blank until the first click
It happens here, for example: https://www.youtube.com/watch?v=-59Bvm3M8Q8
but it's not a thing with multiple editors, I remember it already happened to me many times with gobstones exercises

### PR DESCRIPTION
**Pull request by luchotc**
This PR introduces the following changes:

* By adding the codemirror-autorefresh found on https://codemirror.net/addon/display/autorefresh.js, the glitch when changing editors in multifile editor is fixed. (Resolves #1146)

* Adds support for a codeMirror readOnlyEditor, used both in code and in multifile read_only editor.

* Adds support for creating a discussion of a multifile editor exercise.

* Wrapps every read_only editor in the .mu-read-only-editor class in order to normalize it.

multifile
gobs

### CODE REVIEW

**Comment by flbulgarelli** [comment]
> Wrapps every read_only editor in the .mu-read-only-editor class in order to normalize it.

:confetti_ball:

File: app/assets/javascripts/application/codemirror-autorefresh.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+//From https://codemirror.net/addon/display/autorefresh.js
```
Comment (flbulgarelli) [inline app/assets/javascripts/application/codemirror-autorefresh.js]: Could you place this code in `vendor`, please?

File: domain/app/models/exercise.rb [hunk 2]
```diff
@@ -183,6 +183,13 @@ def limited?
     navigable_parent.limited_for?(self)
   end
 
+  def files_for(current_content)
```
Comment (flbulgarelli) [inline domain/app/models/exercise.rb]: Is there any test for this code? I think you have just refactored an untested method.

Comment (flbulgarelli) [reply domain/app/models/exercise.rb]: I see. I'd **add** a few simple tests in domain for the `files_for` in that case. 

Please proceed, thanks.

**Comment by luchotc** [comment]
@flbulgarelli Ready to re-review :smiley:

**Pull request merged**